tests/unit/s2n_self_talk_nonblocking_test.c

```diff
@@ -58,19 +67,23 @@ int mock_client(struct s2n_test_io_pair *io_pair, uint8_t *expected_data, uint32
 
     /* Receive 10MB of data */
     uint32_t remaining = size;
-    while(remaining) {
+    while(remaining && (start_time - time(0)) < max_wait_time) {
```

Comment:
Maybe a macro would help? Something like

```
#define IS_BEFORE_TIMEOUT(start_time) (((start_time) - time(0)) < 300)
```
Or just a little helper function that returns a bool.

Reply:
Great suggestion. That would clear up the logic.